File: abstract-factory/python/custom_operation.py
from database_factory import DatabaseFactory
from connection import Connection
from command import Command


class CustomOperation:
    _database_factory: DatabaseFactory
    _connection: Connection
    _command: Command

    def __init__(self, database_factory):
        self._database_factory = database_factory
        self._connection = database_factory.create_connection()
        self._command = database_factory.create_command()

    def remove_by_id(self, r_id):
        self._connection.open_connection()
        self._command.execute_command("DELETE...")
        print(f"Removed id: {r_id} from  {self._database_factory}")
        self._connection.close_connection()

File: abstract-factory/go/custom-operation.go
package main

import (
	"fmt"
)

type customOperation struct {
	databaseFactory DatabaseFactory
	connection      Connection
	command         Command
}

func NewCustomOperation(databaseFactory DatabaseFactory) *customOperation {
	return &customOperation{
		databaseFactory: databaseFactory,
		command:         databaseFactory.createCommand(),
		connection:      databaseFactory.createConnection(),
	}
}

func (c *customOperation) removeById(id int) error {
	if err := c.connection.openConnection(); err != nil {
		return fmt.Errorf("error opening Connection: %v", err)
	}
	c.command.executeCommand("Delete ...")
	if err := c.connection.closeConnection(); err != nil {
		return fmt.Errorf("error closing Connection: %v", err)
	}
	return nil
}
